product_configurator/models/product.py

```diff
@@ -269,6 +269,32 @@ def get_config_image_obj(self, value_ids, size=None):
         return img_obj
 
     @api.multi
+    def encode_custom_values(self, custom_values):
+        """ Hook to alter the values of the custom values before creating or writing
+
+            :param custom_values: dict {product.attribute.id: custom_value}
+
+            :returns: list of custom values compatible with write and create
+         """
+        custom_values = custom_values or {}
```

Comment:
As Roel has reminded me in a previous PR, the form var = val or mutable-object is to be avoided.

Best is to use the standard if var == None: var = mutable-object

https://pythonconquerstheuniverse.wordpress.com/2012/02/15/mutable-default-arguments/